sechub-doc/src/test/java/com/mercedesbenz/sechub/docgen/messaging/MessagingAnnotationsCorrectDefinedTest.java

```diff
@@ -0,0 +1,338 @@
+    }
+
+    @Test
+    public void sent_sync_messages_have_at_least_one_sync_receiver() throws Exception {
+        /* prepare */
+
+        /* test */
+        for (MessageID sent : context.messagesSentSynchronous) {
+            if (!context.messagesReceivedSynchronous.contains(sent)) {
+                context.markDiagramGenerationProblem(sent, "sent synchronous, but never received synchronous");
+            }
+        }
+
+        if (context.isNotOkay()) {
+            fail(context.toString());
+        }
+    }
+
+    @Test
+    public void received_sync_messages_may_not_be_received_asynchronous() throws Exception {
+        /* prepare */
+
+        /* test */
+        for (MessageID sent : context.messagesReceivedSynchronous) {
+            if (context.messagesReceivedAsynchronous.contains(sent)) {
+                context.markDiagramGenerationProblem(sent, "received synchronous, but also asynchronous");
+            }
+        }
+
+        if (context.isNotOkay()) {
+            fail(context.toString());
+        }
+    }
+
+    @Test
+    public void received_async_messages_may_not_be_received_synchronous() throws Exception {
+        /* prepare */
+
+        /* test */
+        for (MessageID sent : context.messagesReceivedAsynchronous) {
+            if (context.messagesReceivedSynchronous.contains(sent)) {
+                context.markDiagramGenerationProblem(sent, "received asynchronous, but also synchronous");
+            }
+        }
+
+        if (context.isNotOkay()) {
+            fail(context.toString());
+        }
+    }
+
+    @Test
+    public void all_message_ids_are_sent_at_least_one_time() throws Exception {
+        /* prepare */
+        Set<MessageID> allSentMessages = new HashSet<>();
+        allSentMessages.addAll(context.messagesSentAsynchronous);
+        allSentMessages.addAll(context.messagesSentSynchronous);
+
+        /* test */
+        for (MessageID message : MessageID.values()) {
+            if (!allSentMessages.contains(message)) {
+                if (context.messagesAnswerResultSynchronous.contains(message)) {
+                    /* ignore - it is used as a result for synchronous call, so "sent" and okay */
+                    continue;
+                }
+                context.markDiagramGenerationProblem(message, "is defined, but never sent");
+            }
+        }
+
+        if (context.isNotOkay()) {
+            fail(context.toString());
+        }
+    }
+
+    @Test
+    public void all_message_ids_are_received_at_least_one_time() throws Exception {
+        /* prepare */
+        Set<MessageID> allSentMessages = new HashSet<>();
+        allSentMessages.addAll(context.messagesReceivedAsynchronous);
+        allSentMessages.addAll(context.messagesReceivedSynchronous);
+
+        /* test */
+        for (MessageID message : MessageID.values()) {
+            if (!allSentMessages.contains(message)) {
+                if (context.messagesAnswerResultSynchronous.contains(message)) {
```

Comment:
It was not clear for myself if the changes you proposed would lead to exact same test - but nevertheless I think the test is not very readable (even for myself now).

So I decided to change the test and make it more readable - and remove also the `continue`. Please look at the changes.